### PR TITLE
Publicize: Remove the ability to select personal facebook page from the overlay

### PR DESCRIPTION
**Pull request by enejb**
This change removed the ability for users to select personal pages from the facebook selection. 
Before:
screen shot 2018-07-17 at 10 34 18 am

After:
if the user has 1 page to select
screen shot 2018-07-20 at 8 53 52 am

If the user has no pages to select.
screen shot 2018-07-20 at 8 54 32 am

Another minor fix:
screen shot 2018-07-20 at 12 18 00 pm

This gets display to the user after they connect Facebook but haven't selected the account just yet to which they want to publish to. Before it used to show the users name. 

#### Testing instructions:
Connect your facebook account. 
Does the dialog make sense?

### CODE REVIEW

File: modules/publicize/publicize-jetpack.php [hunk 1]
```diff
@@ -625,8 +604,14 @@ function options_page_facebook() {
 
 					</tbody>
 				</table>
-
-			<?php endif; ?>
+			<?php else: ?>
+				<div>
+					<p><?php _e( sprintf( 'Only Facebook pages are now supported by Publicize. <a href="%s">Learn more about this change</a>', 'https://jetpack.com/support/publicize/facebook' ), 'jetpack' ); ?></p>
```
Comment (mdawaffe) [inline modules/publicize/publicize-jetpack.php]: I think we should include this note in the page selection dialog as well as here in the dialog when there are no pages.

Also, the text here makes it sound like Publicize is limited (when really it's Facebook). If I saw the current message, I would probably look for a different plugin to automate posting to my Facebook Profile (not realizing it was no longer possible).

I don't want to reword it to blame Facebook and make us look better :) but making it clear that this is limitation of Facebook's will help people understand the situation better.

Maybe something like the following?

> Facebook only supports Publicize for Facebook Pages. Publicize for Facebook Profiles is not possible. Learn More about Publicize for Facebook.

I'm sure +editorial will be able to help as well.

Comment (mdawaffe) [reply modules/publicize/publicize-jetpack.php]: This i18n is broken. `_e()` can't contain `sprintf()`. `printf( __() )` will work better.

File: modules/publicize/publicize-jetpack.php [hunk 2]
```diff
@@ -625,8 +604,14 @@ function options_page_facebook() {
...
+					<p><?php _e( sprintf( '<a class="button" href="%s" target="_blank">Create a Facebook page</a> to get started.', 'https://www.facebook.com/pages/creation/'), 'jetpack' ); ?></p>
```
Comment (mdawaffe) [inline modules/publicize/publicize-jetpack.php]: i18n

**Comment by enejb** [comment]
@mdawaffe This is the other PR. 
That tells the user that their current connection is broken. 
https://github.com/Automattic/jetpack/pull/9938 and is not working as expected.

**Comment by benhuberman** [comment]
@enejb on the wording front, maybe we can streamline a bit (on the "no pages to select" screen):
> Facebook supports Publicize connections to Facebook Pages, but not to Facebook Profiles.
The rest of the copy looks fine to me.

Question: on the updated screen for people with pages to select, would it make sense to add a quick note somewhere (maybe near the bottom?) along the same lines as the message I propose above? I'm just thinking it might prevent confusion from users who don't understand _why_ they don't see their profile listed there.

**Comment by enejb** [comment]
@benhuberman Thanks for the suggestion I think that makes perfect sense. I will add a note a note for users that see only pages.

**Comment by enejb** [comment]
Screen shots updated.

**Comment by mdawaffe** [comment]
Something is not working correctly. I'm not sure if I set something up wrong.

1. I used https://jurassic.ninja/create?jetpack-beta&branch=update/facebook-personal-sharing-ui&shortlived&wp-debug-log to create a test site.
2. I connected Jetpack.
3. I went to Settings -> Sharing and clicked "Connect" for Facebook.
4. I saw the page select dialog. It showed only pages; it did not show my profile.
5. I selected a page, and clicked OK.
6. The info in Settings -> Sharing now says "Connected as: Michael Adams" (which is the name of my profile not the name of the page).
7. The checkbox in the post editor also says "Facebook: Michael Adams".
8. Looking at the connection data in the database, `is_page` is set to false.

So I used the UI to select a page, but I still ended up being connected to my profile.

**Comment by enejb** [comment]
Thanks @mdawaffe Will investigate this more.

**Comment by enejb** [comment]
I just tried replicating this. 
Can you tell me what the site's `Jetpack_Options::get_option( 'publicize_connections' );` 
looks like?

**Comment by enejb** [comment]
I just tried replicating this and I wasn't able to replicate it :/

**Comment by mdawaffe** [comment]
> Can you tell me what the site's Jetpack_Options::get_option( 'publicize_connections' );
looks like?

```
$ wp shell
wp> Jetpack_Options::get_option( 'publicize_connections' )
=> array(1) {
  ["facebook"]=>
  array(1) {
    [20525692]=>
    array(9) {
      ["user_id"]=>
      int(1)
      ["provider"]=>
      string(21) "facebook:249643311490"
      ["issued"]=>
      string(19) "2018-07-20 19:38:59"
      ["expires"]=>
      string(19) "0000-00-00 00:00:00"
      ["external_id"]=>
      string(7) "3708111"
      ["external_name"]=>
      string(0) ""
      ["external_display"]=>
      string(13) "Michael Adams"
      ["type"]=>
      string(6) "access"
      ["connection_data"]=>
      array(5) {
        ["id"]=>
        string(8) "20525692"
        ["token_id"]=>
        string(8) "20671501"
        ["blog_id"]=>
        string(9) "149274350"
        ["user_id"]=>
        int(1)
        ["meta"]=>
        array(5) {
          ["external_user_id"]=>
          int(1)
          ["facebook_page"]=>
          int(161041690648078)
          ["follower_count"]=>
          int(330)
          ["link"]=>
          string(197) "https://www.facebook.com/app_scoped_user_id/[REDACTED]/"
          ["display_name"]=>
          string(13) "Michael Adams"
        }
      }
    }
  }
}
```

On WordPress.com, the data we store in `publicize_account_info` is:

```
array (
  'fb' => 
  array (
    20671501 => 
    array (
      'id' => 3708111,
      'link' => 'https://www.facebook.com/app_scoped_user_id/[REDACTED]/',
      'display' => 'Michael Adams',
      'is_page' => false,
    ),
  ),
);
```

**Comment by enejb** [comment]
Thanks for the info.

**Comment by enejb** [comment]
And everything work as expected for with the current version of Jetpack? 
if you select the same Facebook page?

I wonder if there is a difference between the test facebook account that I am using and the one that you have?

**Comment by enejb** [comment]
I wonder how we could prevent this from happening? Any suggestions? 
is there any indication that the page the user is about to select broken unpublished. 

You could check by trying to connect but then never selecting the account. 
You should be able to take a look at the `Jetpack_Options::get_option( 'publicize_connections' )` then and see all the possible accounts. As an option on that array.

**Comment by mdawaffe** [comment]
> I wonder how we could prevent this from happening?

This is what seems to happen:
1. We forge the connection.
2. We show the user a list of their Pages.
3. The user selects which Page.
4. We check to see if that Page is published (and a few other things).
   a. If it is, we set the Page token.
   b. If it isn't we do not set the Page token.
5. If the page token is not set, we default to displaying info from the Profile (and publicizing to it).

I haven't looked too deeply into things to confirm that, but see the handler for `jetpack.setPublicizeOptions` (WP.com's wp-content/mu-plugins/jetpack/class.jetpack-xmlrpc-server.php) and WP.com's `Keyring_Service_Facebook::get_fb_pages()`).

When we display the Pages to the user, we should probably show all Pages but disable the radio buttons for Pages we can't publish to.

On the Jetpack side, I don't see anything in `Jetpack_Options::get_option( 'publicize_connections' )` that differentiates between unpublished and published pages.

This whole flow, though, makes me think it's still possible to set up Publicize to a Profile (accidentally or on purpose). There may be two ways to do this:
1. Have only unpublished Pages, and the code seems to fall back to your Profile.
2. Or, as you suggested, if the user abandons the connection flow right before selecting a page, will Publicize ("Facebook: Connecting Facebook...") post things to the profile?

If either works, I guess we should just make sure we never try to Publicize anything unless a Page really is selected. (I know you're already working on this sort of thing - thanks!)

**Comment by enejb** [comment]
True. 
I can try to address this better when I merge this PR and  https://github.com/Automattic/jetpack/pull/9938

Do you mind approving it if you think this is a good plan? 

Also I feel like if the user has not selected anything we should try to just remove the Facebook connection part completely. 
This is something that I plan on fixing in a future PR.

**Comment by zinigor** [comment]
Ported to `branch-6.3` in 871a3bf3d.
